VTUL/iawa
# LIBTD-1700: Restrict the access to sidekiq to admin users only

**JIRA Ticket**: (https://webapps.es.vt.edu/jira/browse/LIBTD-1700) (:star:)

# What does this Pull Request do? (:star:)
This PR sets restriction on using the web interface to monitor the sidekiq background jobs so that only admin users can see, change or stop the jobs. We use Devise authentication to limit the access to the sidekiq route.

# What's the changes? (:star:)

*  in config/routes.rb, the route to sidekiq is wrapped around by Devise authentication constraint to admin users.

# How should this be tested?

* This PR should be tested using LIBTD-1700 branch either in development or production mode;
* You can set yourself up as admin or non-admin user by manipulating the ansible/local_files/admin_list.txt file; 
* First sign up as a non-admin user and confirm that iawa_site/sidekiq is not accessible;
* Then sign up as an admin user and confirm that iawa_site/sidekiq is accessible.

# Interested parties
Tag (@shabububu ) interested parties

(:star:) Required fields
